### ISSUE TITLE
BABE implementation

### ISSUE THREAD

**Issue by kwanCCC** [zeeka-network/doc#2]
First important thing is _VRF_, We will use the Schnorrkel VRF and sr25519 signatures. But I'm not sure about the BLS because most BLS makes hashing-to-the-curve slow.

- [ ] Digest in Block Header (Primary, Secondary)
- [ ] The API of VRF(epoch_randomness, slot_num)
- [ ] Distribute Clock system
- [ ] block proposing 
- [ ] block verification

**Comment by keyvank** [comment]
@kwanCCC Do you want to implement VRFs from scratch? I think a good strategy is to make abstract `traits` of signatures, VRFs and all other crypto primitives (Like what I did here: https://github.com/zeeka-network/bazuka/blob/master/src/crypto.rs ) and then use prewritten libraries to implement these traits. This way we can make a MVP faster, then if we figured out we don't want to use those VRF libraries, we can just implement those traits again ourselves. I was going to using https://github.com/witnet/vrf-rs library as our VRF backend. It uses secp256k1 curves. Is that ok? Or am I missing something?

**Comment by kwanCCC** [comment]
@keyvank 
No,I don't want implement VRFs from scratch. And I found [schnorrkel](https://github.com/w3f/schnorrkel), but only support Ristretto compressed Ed25519 points. The task _VRF_ which one hasn't clear description described above. The goal of _VRF_ is make it as part of _KeyStore_ which works as a module in whole system then we can use _KeyStore_ to sign a value using VRF, verify a proof.

**Comment by kwanCCC** [comment]
```rust
pub trait CryptoStore {
  fn generate_new_keypair();
  fn sign_with(msg);
  fn vrf_sign();
  .......
}
```
The Api of _KeyStore_ maybe like this or some more methods.
